### ISSUE TITLE
[feature] conan 2.0 - Better indications that user needs to login to remote

### ISSUE THREAD

**Issue by jsallay** [conan-io/conan#12574]
I have a case where I have a private conan remote that a user must log into.  In conan 1.0, if I do a command such as `conan install pkg/version@` it will ask me to login to a remote if I haven't done so yet.  In conan 2.0, `conan install --requires pkg/version` returns a message saying `Unable to find 'pkg/version' in remotes`.  Other commands like `conan search` exhibit similar behavior in conan 2.0.

I have found that a user can type `conan remote list-users` to see if they are logged in.  However, this isn't their first instinct when they can't find the package of interest.  They first think that they must have misspelled the package name or used the wrong version number.  It would be really nice if commands like `search` and `install` could "know" that a user needs to login to be able to use it.

I realize that this is somewhat complicated because some remote don't need a login so we can't just assume that every remote needs one.  I would propose adding a new field to the `remotes.json`, something like `requiresLogin`.  If the field is not present, then everything can function how it does today.  If it is present and set to `true`, then commands like `search` would be able to print out something like
```
# Proposed
remote:
   Cannot search remote without login.
```

Instead of
```
# Current
remote:
  There are no matching recipe references
```

The install command could do something similar.  If it can't find a recipe, then it could print a list of the remotes that are marked as requiring a login, but aren't logged in.

We work in docker containers and so it is fairly common for someone to spin up a container and forget to login.  I originally had the idea of adding the login command to .bashrc, but that causes a host of other issues (e.g. the user has to be connected to the internet in order to start a fresh docker container).

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @jsallay 

Trying to understand a bit better, because I think this issue has been requested before, and it couldn't be implemented because of the way security is managed in the server side.

I am trying in conan 2.0, and if I have my private remote and I do ``conan search * -r=myremote`` then it will also force the auth with user/password in the same way it is required for a ``conan install``. It will not report "There are no matching...". Could you please check it and clarify?

**Comment by jsallay** [comment]
I have a fresh ubuntu20.04 container running beta5.  The output of `conan search "*" -r=remote` is:
```
remote:
  There are no matching recipe references
```

I am using artifactory to store my packages.  I have two remote repos.  A public one and a private one.  I was using a virtual repo to group them together, which I thought could be part of the problem.  However, I started another fresh container.  Set my remote to point only to the private repo, and still have the same issue.

**Comment by jsallay** [comment]
I looked at my artifactory settings and it looks like that is the issue.  I was able to change a setting and now it asks for a login on the private repo when I do a search.

The issue still exists for the virtual repo, but I don't think I need that anymore.  If I remember correctly, in conan 1.0 most commands could only specify a single remote.  If I had two remotes that I wanted to draw packages from there wasn't an easy way to handle that.  It looks like I can now specify `-r` multiple times on the command line.  So I don't need the virtual repo any more.

You guys are on the ball.  Thanks.
